lib/galaxy/tools/model_operation_macros.xml

```diff
@@ -4,6 +4,38 @@
             class="ModelOperationToolAction"/>
     </xml>
     <token name="@QUOTA_USAGE_NOTE@">This tool will create new history datasets copied from your input collections but your quota usage will not increase.</token>
+    <token name="@CROSS_PRODUCT_INTRO@"><![CDATA[
+This tool organizes two dataset lists so that Galaxy's normal collection processing produces
+an all-vs-all style analyses of the initial inputs when applied to the outputs of this tool.
+
+While a description of what it does standalone is technical and math heavy, how
+it works within an ad-hoc analysis or workflow can be quite straight forward and hopefully is easier
+to understand. For this reason, the next section describes how to use this tool in context and
+the technical details follow after that. Hopefully, the "how it works" details aren't nessecary to
+understand the "how to use it" details of this tool - at least for simple things.
+]]>
+</token>
+    <token name="@GALAXY_DOT_PRODUCT_SEMANTICS@"><![CDATA[
+
+This tool can be used in and out of workflows, but workflows will be used to illustrate the ordering of
+tools and connections between them. Imagine a tool that compares two individual datasets and how
+that might be connected to list inputs in a workflow. This simiple case is shown below:
+
+.. image:: ${static_path}/images/tools/collection_ops/dot_product.png
+  :alt: The Dot Product of Two Collections
+  :width: 500
+
+In this configuration - the two datasets will be matched and compared element-wise. So the first dataset
+of "Input List 1" will be compared to the first dataset in "Input List 2" and the resulting
+dataset will be the first dataset in the output list generated using this comparison tool. In this configuration
+the lists need to have the same number of elements and ideally matching element identifiers.
+
+This matching up of elements is a very natural way to "map" an operation (or in Galaxy parlance, a tool)
+over two lists. However, sometimes the desire is to compare each element of the first list to each element of the
```

Comment:
```suggestion
This matching up of elements is a very natural way to perform a "map over" operation, which in Galaxy means to map an operation over the elements of multiple lists. However, sometimes the desire is to compare each element of the first list to each element of the
```

I think I would want to establish the term "map over" (and quote it) since that's what (I think) we use when we're talking about these operations. It has probably entered the conversation in chats and the help forum at this point.

Feel free to dismiss this though, it's great that we finally put it into writing.

Reply:
I have opened a PR that I think would enable us to do this well in dev https://github.com/galaxyproject/galaxy/pull/18722.